server/apiv2/handlers/keyspace.go

```diff
@@ -0,0 +1,346 @@
+		}
+	} else {
+		// Scanned limit + 1 keyspaces, there is next page, all but last are results.
+		resultKeyspaces = make([]*KeyspaceMeta, len(scanned)-1)
+		for i := range resultKeyspaces {
+			resultKeyspaces[i] = &KeyspaceMeta{scanned[i]}
+		}
+		// Also set next_page_token here.
+		resp.NextPageToken = strconv.Itoa(int(scanned[len(scanned)-1].Id))
+	}
+	resp.Keyspaces = resultKeyspaces
+	c.IndentedJSON(http.StatusOK, resp)
+}
+
+// UpdateConfigParams represents parameters needed to modify target keyspace's configs.
+// NOTE: This type is exported by HTTP API. Please pay more attention when modifying it.
+// A Map of string to string pointer is used to differentiate between json null and "",
+// which will both be set to "" if value type is string during binding.
+type UpdateConfigParams struct {
+	Config map[string]*string `json:"config"`
+}
+
+// UpdateKeyspaceConfig updates target keyspace's config.
+// This is a custom action that supports PATCH semantics and uses the JSON merge patch
+// format and processing rules.
+// @Tags keyspaces
+// @Summary Update keyspace config.
+// @Param name path string true "Keyspace Name"
+// @Param body body UpdateConfigParams true "Update keyspace parameters"
+// @Produce json
+// @Success 200 {object} KeyspaceMeta
+// @Failure 500 {string} string "PD server failed to proceed the request."
+// Router /keyspaces/{name}/update-config [post]
+func UpdateKeyspaceConfig(c *gin.Context) {
+	svr := c.MustGet("server").(*server.Server)
+	manager := svr.GetKeyspaceManager()
+	name := c.Param("name")
+	configParams := &UpdateConfigParams{}
+	err := c.BindJSON(configParams)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, errs.ErrBindJSON.Wrap(err).GenWithStackByCause())
+		return
+	}
+	mutations := getMutations(configParams.Config)
+	meta, err := manager.UpdateKeyspaceConfig(name, mutations)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+		return
+	}
+	c.IndentedJSON(http.StatusOK, &KeyspaceMeta{meta})
+}
+
+// getMutations converts a given JSON merge patch to a series of keyspace config mutations.
+func getMutations(patch map[string]*string) []*keyspace.Mutation {
+	mutations := make([]*keyspace.Mutation, 0, len(patch))
+	for k, v := range patch {
+		if v == nil {
+			mutations = append(mutations, &keyspace.Mutation{
+				Op:  keyspace.OpDel,
+				Key: k,
+			})
+		} else {
+			mutations = append(mutations, &keyspace.Mutation{
+				Op:    keyspace.OpPut,
+				Key:   k,
+				Value: *v,
+			})
+		}
+	}
+	return mutations
+}
+
+// EnableKeyspace enables target keyspace.
+// @Tags keyspaces
+// @Summary Enable keyspace.
+// @Param name path string true "Keyspace Name"
+// @Produce json
+// @Success 200 {object} KeyspaceMeta
+// @Failure 500 {string} string "PD server failed to proceed the request."
+// Router /keyspaces/{name}/enable [post]
+func EnableKeyspace(c *gin.Context) {
+	updateKeyspaceState(c, keyspacepb.KeyspaceState_ENABLED)
```

Comment:
Should we use PUT for updating the existingd state?

Reply:
Changing the keyspace state may cause some side effects other than the state change,
that's why I think POST + custom action is more suited instead of PUT.
A example of this can be seen at GCE instance api's [start](https://cloud.google.com/compute/docs/reference/rest/v1/instances/start)/stop/suspend